tests/pyfunc/test_model_export_with_class_and_artifacts.py

```diff
@@ -1273,7 +1273,7 @@ def test_functional_python_model_list_to_list(tmp_path):
     assert loaded_model.predict([{"a": "x"}, {"a": "y"}]) == ["x", "y"]
 
 
-def list_to_list_pep585(x: list[str]) -> list[str]:
+def list_to_list_pep585(x: List[str]) -> List[str]:
     return x
```

Comment:
This is to make sure type inference works fine with `list[str]` as well.

Reply:
Does this PR remove the input conversion using type hints?

Reply:
It didn't, but when I run this test locally `type_hint == List[str]` fails for this, so I get `INFO mlflow.types.utils: Unsupported type hint: list[str], skipping schema inference`

Reply:
Ok I tried python 3.8 and this test passes, it fails on python 3.9